### ISSUE TITLE
Broken with Vaadin 23.0.0.beta4 

### ISSUE THREAD

**Issue by simasch** [mvysny/karibu-testing#101]
It seems that Karibu Testing is broken with Vaadin 23.0.0.beta4

```
java.lang.NoClassDefFoundError: com/vaadin/shrinkwrap/VaadinCoreShrinkWrap

	at com.github.mvysny.kaributools.VaadinVersion$get$2.invoke(VaadinVersion.kt)
	at com.github.mvysny.kaributools.VaadinVersion$get$2.invoke(VaadinVersion.kt:83)
	at kotlin.SafePublicationLazyImpl.getValue(LazyJVM.kt:107)
	at com.github.mvysny.kaributools.VaadinVersion.getGet(VaadinVersion.kt:83)
	at com.github.mvysny.kaributesting.v10.MockVaadin.setup(MockVaadin.kt:78)
	at ch.x.vaadin.test.AbstractKaribuTest.setup(AbstractKaribuTest.java:48)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptLifecycleMethod(TimeoutExtension.java:126)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptBeforeEachMethod(TimeoutExtension.java:76)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeMethodInExtensionContext(ClassBasedTestDescriptor.java:506)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$synthesizeBeforeEachMethodAdapter$21(ClassBasedTestDescriptor.java:491)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeEachMethods$3(TestMethodTestDescriptor.java:171)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeMethodsOrCallbacksUntilExceptionOccurs$6(TestMethodTestDescriptor.java:199)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeMethodsOrCallbacksUntilExceptionOccurs(TestMethodTestDescriptor.java:199)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeEachMethods(TestMethodTestDescriptor.java:168)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
```

**Comment by joheriks** [comment]
As of 23.0.0.beta4, shrinkwrap is no longer used and [these classes have been removed.](https://github.com/vaadin/platform/pull/2751). To retrieve the plaform version, you can instead use https://github.com/vaadin/flow/blob/master/flow-server/src/main/java/com/vaadin/flow/server/Platform.java

**Comment by knoobie** [comment]
Not sure where to ask this; that's why am asking here @joheriks :) Was there any reason to use this approach (reading files and such) instead of creating a manifest entry and read the version with `Platform.class.getPackage().getImplementationVersion()` now that flow and the platform share the same version?

**Comment by joheriks** [comment]
Technically they don't share the version as the patchlevel and qualifier may diverge. But for purposes where the Flow version (which shares major and minor with platform) is enough, using  `com.vaadin.flow.server.Version` is an alternative.

**Comment by mvysny** [comment]
Blocked by https://github.com/mvysny/karibu-tools/issues/4

**Comment by mvysny** [comment]
Fix released in Karibu-Testing 1.3.11.

**Comment by simasch** [comment]
Thank you @mvysny
